modules/generator/generator.go

```diff
@@ -38,6 +38,12 @@ const (
 	// We use a safe default instead of exposing to config option to the user
 	// in order to simplify the config.
 	ringNumTokens = 256
+
+	// NoGenerateMetricsContextKey is used in request contexts/headers to signal to
+	// the metrics generator that it should not generate metrics for the spans
+	// contained in the requests. This is intended to be used by clients that send
+	// requests for which span metrics have already been generated elsewhere.
+	NoGenerateMetricsContextKey = ingest.NoGenerateMetricsContextKey
```

Comment:
Out of curiosity, what's the reason for redefining `NoGenerateMetricsContextKey` in each pakage?

Reply:
My main reason for this was to keep dependencies "logical", i.e. a client pushing data to the generator needs to import the generator package to figure out what special value to use when skipping data, etc. If you think it's better I'm happy to make them all import `ingest` directly instead.

Reply:
With the latest changes there are only two places left that define this key (but now independently), and only the generator package exports it (so clients can use that value by importing the package).